### PR TITLE
Build: Add test for rocksdb availability and minimum version

### PR DESCRIPTION
**Pull request by EchterAgo**
This will use the system `rocksdb` if it is available and meets the minimum version requirement, otherwise it will use the prebuilt static library.

### CODE REVIEW

File: config.tests/rocksdb/main.cpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        } else if (version1[i] < version2[i]) {
+            return false;
+        }
+    }
+    return true;
+}
+
+static_assert(compareVersion(version, minimumVersion, sizeof(version) / sizeof(version[0])));
+
+int main() {}
```
Comment (cculianu) [inline config.tests/rocksdb/main.cpp:21]: You may want to return 0 here unconditionally.  I'm not sure what `qtCompileTest()` expects but I would be surprised if it does not expect 0 return status...

Or does it just check if file compiles and doesn't care about return status?

Comment (EchterAgo) [reply config.tests/rocksdb/main.cpp:21]: Yea, it only checks compilation. I initially had this as a runtime test, but that didn't work.

**Comment by cculianu** [comment]
Thanks so much for this. This gets rid of that `features=` thing. Yay!

I'll have to check how this affects my static builds -- if it does I may want to add a features thing to force static later.  But this is fine.  Thank you!
